File: Adhiti_Shaili/Day 13/Ways to Sum to N.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution
{
  public:
    // function to count the total  
    // number of ways to sum up to 'N'
    int mod=pow(10,9)+7;

    int countWays(int arr[], int m, int N) 
    { 
      vector<int> dp(N+1,0);
      dp[0]=1;
      for(int i=1;i<=N;i++)
      {
          for(int j=0;j<m;j++)
          {
              if(i>=arr[j])
              {
                  dp[i]=(dp[i]%mod+dp[i-arr[j]]%mod)%mod;
              }
          }
      }
      return dp[N];
      
    }
}; 

File: sanket-dalvi/Day13-14/waysToSumToN.py
class Solution:
    def countWays(self, arr, m, n):
        nums = [1]+[0]*n
        for i in range(1,n+1):
            for j in range(len(arr)):
                if i >= arr[j]:
                    nums[i] += nums[i-arr[j]]
        return nums[-1]%1000000007

# Test
print(Solution().countWays([1,5,6], 3, 7))